sources/engine/Stride.VirtualReality/OculusOVR/OculusOverlay.cs

```diff
@@ -36,7 +39,11 @@ public OculusOverlay(IntPtr ovrSession, GraphicsDevice device, int width, int he
                 }
 
                 textures[i] = new Texture(device);
-                textures[i].InitializeFromImpl(new Texture2D(ptr), false);
+                unsafe
+                {
+                    var tex = new ID3D11Texture2D((void**)ptr);
+                    textures[i].InitializeFromImpl(new ComPtr<ID3D11Texture2D>(&tex), false);
```

Comment:
This also won't work as intended, for almost the same reason.

For context, how our struct wrappers work is they take a pointer to `&this`, which when used as a pointer maps to what DirectX has given us. I [explained it to you in our Discord server](https://discord.com/channels/521092042781229087/607634593201520651/904837269075157002) before, but paraphrasing it here for others' reference:
> let's say the variable [tex] is at address `0x1` and [D3D11] returns address `0x2` as the [texture]'s address, with some internal state holding the [texture] info at `0x3` (i.e. `0x2+1`). when you dereference `0x2` into `0x1`, when [InitializeFromImpl] tries to get `0x1+1` you get a segfault because that state doesn't exist - `&this` is now `0x1` instead of `0x2`. 

You're probably looking for:
```suggestion
                    var tex = new ComPtr<ID3D11Texture2D>((ID3D11Texture2D*)ptr);
                    textures[i].InitializeFromImpl(tex, false);
```